datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java

```diff
@@ -407,7 +412,14 @@ public SdxGenerateImageCatalogResponse generateImageCatalog(@ResourceName String
     @Override
     @CheckPermissionByAccount(action = AuthorizationResourceAction.CREATE_DATALAKE)
     public SdxDefaultTemplateResponse getDefaultTemplate(SdxClusterShape clusterShape, String runtimeVersion, String cloudPlatform) {
-        return sdxService.getDefaultTemplate(clusterShape, runtimeVersion, cloudPlatform);
+        return sdxRecommendationService.getDefaultTemplateResponse(clusterShape, runtimeVersion, cloudPlatform);
+    }
+
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.CREATE_DATALAKE)
```

Comment:
another permission check should be added on this method to check if user has get right over credential